[python-pip] Module in `cdktf.json` is not being installed correctly, with 'cdktf get' command

### Expected Behavior

By referring following: https://github.com/cdktf/docker-on-aws-ecs-with-terraform-cdk-using-typescript/blob/main/infrastructure/cdktf.json

I've setup `cdktf.json` as:
```
{
  "language": "python",
  "app": "python3 ./main.py",
  "projectId": "5f6d6bcb-daa6-4dae-9ae1-d02cd37c9f08",
  "sendCrashReports": "true",
  "terraformProviders": [],
  "terraformModules": [
    "terraform-aws-modules/vpc/aws@5.0.0",
    "terraform-aws-modules/rds/aws@5.0.0"
  ],
  "codeMakerOutput": "imports",
  "context": {}
}
```

And this is `main.py` file
```
from imports.terraform_aws_modules.aws import Vpc
from imports.terraform_aws_modules.aws import Rds


class MyStack(TerraformStack):
...
```

It should success when running `cdktf diff`.

### Actual Behavior

Shows failure with following log
```
$ cdktf diff
...
[2023-07-22T20:58:49.628] [ERROR] default - Traceback (most recent call last):
  File "/Users/kination/workspace/private/rz-dev/rz-iac/./main.py", line 6, in <module>
    from imports.terraform_aws_modules.aws import Rds
ImportError: cannot import name 'Rds' from 'imports.terraform_aws_modules.aws' (/Users/kination/workspace/private/rz-dev/rz-iac/
ERROR: cdktf encountered an error while synthesizing

Synth command: python3 ./main.py
Error:         non-zero exit code 1

Command output on stderr:

    Traceback (most recent call last):
      File "/Users/kination/workspace/private/rz-dev/rz-iac/./main.py", line 6, in <module>
        from imports.terraform_aws_modules.aws import Rds
    ImportError: cannot import name 'Rds' from 'imports.terraform_aws_modules.aws' (/Users/kination/workspace/private/rz-dev/rz-iac/imports/terraform_aws_modules/aws/__init__.py)
```

Seems only vpc module has been installed, and rds has not.


### Steps to Reproduce

1. Run `cdktf init --template=python-pip --local`
2. Add following modules
```
  "terraformModules": [
    "terraform-aws-modules/vpc/aws@5.0.0",
    "terraform-aws-modules/rds/aws@5.0.0"
  ],
```
3. Run `cdktf get`
4. After install, run `cdktf diff`


### Versions

language: python
cdktf-cli: 0.0.0
node: v16.14.2
cdktf: 0.0.0 (master branch)
constructs: 10.2.69
jsii: 1.85.0
terraform: 1.4.2
arch: arm64
os: darwin 22.5.0 (MacOS M1)
python: Python 3.9.1
pip: null
pipenv: pipenv, version 2023.3.20

### Providers

_No response_

### Gist

_No response_

### Possible Solutions

_No response_

### Workarounds

_No response_

### Anything Else?

_No response_

### References

_No response_

### Help Wanted

- [X] I'm interested in contributing a fix myself

### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,3 +1,11 @@
+## 0.20.0 (Unreleased)
+
+**Breaking changes**
+
+### Module names / import paths changing
+
+To resolve [\#3026](https://github.com/hashicorp/terraform-cdk/issues/3026) we improved the logic for generating names for [Terraform Modules](https://developer.hashicorp.com/terraform/cdktf/concepts/modules). This makes the default module names nicer, but also leads to breaking changes for Python, C#, and Java. When updating CDKTF, please generate the new bindings with `cdktf get` after updating the CLI and update your module imports.
+
 ## 0.19.1
 
 ### feat

--- a/examples/python/aws-eks/cdktf.json
+++ b/examples/python/aws-eks/cdktf.json
@@ -9,16 +9,8 @@
     }
   ],
   "terraformModules": [
-    {
-      "name": "vpc",
-      "source": "terraform-aws-modules/vpc/aws",
-      "version": "2.77.0"
-    },
-    {
-      "name": "eks",
-      "source": "terraform-aws-modules/eks/aws",
-      "version": "~> 15.0"
-    }
+    "terraform-aws-modules/vpc/aws@5.0.0",
+    "terraform-aws-modules/rds/aws@5.0.0"
   ],
   "codeMakerOutput": "imports"
 }

--- a/examples/python/aws-eks/cdktf.json
+++ b/examples/python/aws-eks/cdktf.json
@@ -9,9 +9,16 @@
     }
   ],
   "terraformModules": [
-    "terraform-aws-modules/vpc/aws@5.0.0",
-    "terraform-aws-modules/rds/aws@5.0.0"
+    {
+      "name": "vpc",
+      "source": "terraform-aws-modules/vpc/aws",
+      "version": "2.77.0"
+    },
+    {
+      "name": "eks",
+      "source": "terraform-aws-modules/eks/aws",
+      "version": "~> 15.0"
+    }
   ],
   "codeMakerOutput": "imports"
 }
-

--- a/examples/python/aws/main.py
+++ b/examples/python/aws/main.py
@@ -3,11 +3,11 @@
 
 from constructs import Construct
 from cdktf import App, TerraformStack
+from imports.aws.iam_role import IamRole
+from imports.aws.lambda_function import LambdaFunction
 from imports.aws.provider import AwsProvider
 from imports.aws.sns_topic import SnsTopic
-from imports.terraform_aws_modules.aws import Vpc
-from imports.aws.lambda_function import LambdaFunction
-from imports.aws.iam_role import IamRole
+from imports.vpc import Vpc
 
 
 class MyStack(TerraformStack):

--- a/packages/@cdktf/commons/src/config.ts
+++ b/packages/@cdktf/commons/src/config.ts
@@ -143,7 +143,7 @@ export class TerraformModuleConstraint
       toProcess = prefixMatch[2];
     }
 
-    // strip off any protocl
+    // strip off any protocol
     const protocolMatch = toProcess.match(/^([a-zA-Z]*):\/\/(.*)/);
     if (protocolMatch) {
       toProcess = protocolMatch[2];

--- a/packages/@cdktf/commons/src/construct-maker-target.test.ts
+++ b/packages/@cdktf/commons/src/construct-maker-target.test.ts
@@ -1,0 +1,80 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import { Language, TerraformModuleConstraint } from "./config";
+import { ConstructsMakerModuleTarget } from "./construct-maker-target";
+
+describe("ConstructsMakerModuleTarget", () => {
+  describe.each([
+    {
+      fqn: "cloudposse/label/null",
+      names: {
+        [Language.TYPESCRIPT]: "cloudposse.null",
+        [Language.PYTHON]: "cloudposse.label.null",
+        [Language.JAVA]: "cloudposse.label.null",
+        [Language.CSHARP]: "cloudposse.label.null",
+        [Language.GO]: "label",
+      },
+    },
+    {
+      fqn: "terraform-google-modules/project-factory/google",
+      names: {
+        [Language.TYPESCRIPT]: "terraform_google_modules.google",
+        [Language.PYTHON]: "terraform_google_modules.project_factory.google",
+        [Language.JAVA]: "terraform_google_modules.project_factory.google",
+        [Language.CSHARP]: "terraform_google_modules.project_factory.google",
+        [Language.GO]: "project_factory",
+      },
+    },
+    {
+      fqn: "terraform-aws-modules/vpc/aws@5.0.0",
+      names: {
+        [Language.TYPESCRIPT]: "terraform_aws_modules.aws",
+        [Language.PYTHON]: "terraform_aws_modules.vpc.aws",
+        [Language.JAVA]: "terraform_aws_modules.vpc.aws",
+        [Language.CSHARP]: "terraform_aws_modules.vpc.aws",
+        [Language.GO]: "vpc",
+      },
+    },
+    {
+      fqn: "terraform-aws-modules/eks/aws//modules/self-managed-node-group",
+      names: {
+        [Language.TYPESCRIPT]: "terraform_aws_modules.aws.eks.modules",
+        [Language.PYTHON]:
+          "terraform_aws_modules.eks.aws.modules.self_managed_node_group",
+        [Language.JAVA]:
+          "terraform_aws_modules.eks.aws.modules.self_managed_node_group",
+        [Language.CSHARP]:
+          "terraform_aws_modules.eks.aws.modules.self_managed_node_group",
+        [Language.GO]: "self_managed_node_group",
+      },
+    },
+    {
+      fqn: {
+        name: "my-local-module",
+        source: "./path/to/local/terraform/module",
+      },
+      names: {
+        [Language.TYPESCRIPT]: "my-local-module",
+        [Language.PYTHON]: "my_local_module",
+        [Language.JAVA]: "my_local_module",
+        [Language.CSHARP]: "my_local_module",
+        [Language.GO]: "my_local_module",
+      },
+    },
+  ])("#srcMakName %#", ({ fqn, names }) => {
+    it.each(Object.entries(names))(
+      `expect name for ${JSON.stringify(fqn)} in %s to be %s`,
+      (language, name) => {
+        const constraint = new TerraformModuleConstraint(fqn as any);
+        const target = new ConstructsMakerModuleTarget(
+          constraint,
+          language as Language
+        );
+        expect(target.srcMakName).toBe(name);
+      }
+    );
+  });
+});

--- a/packages/@cdktf/commons/src/construct-maker-target.ts
+++ b/packages/@cdktf/commons/src/construct-maker-target.ts
@@ -87,6 +87,7 @@ export class ConstructsMakerModuleTarget extends ConstructsMakerTarget {
       case Language.JAVA:
       case Language.CSHARP:
       case Language.PYTHON:
+        return this.simplifiedFqn;
       default:
         return this.simplifiedName;
     }
@@ -109,6 +110,10 @@ export class ConstructsMakerModuleTarget extends ConstructsMakerTarget {
     return (
       this.namespace?.replace(/\//gi, ".").replace(/-/gi, "_") ?? this.name
     );
+  }
+
+  protected get simplifiedFqn(): string {
+    return this.fqn.replace(/\//gi, ".").replace(/-/gi, "_");
   }
 }
 

